src/pages/RepoPage/PullsTab/PullsTable/createPullsTableData.tsx

```diff
@@ -1,34 +1,67 @@
 import isArray from 'lodash/isArray'
 
 import { Pull } from 'services/pulls/usePulls'
+import { formatSizeToString } from 'shared/utils/bundleAnalysis'
 import TotalsNumber from 'ui/TotalsNumber'
 
-import Coverage from './Coverage'
+import Title from './Title'
 
-import Title from '../shared/Title'
+export const ErroredUpload = () => (
+  <p>
+    Upload: <span aria-label="Errored upload">&#x274C;</span>
+  </p>
+)
+export const PendingUpload = () => (
+  <p>
+    Upload: <span aria-label="Pending upload">&#x23F3;</span>
+  </p>
+)
 
 export const createPullsTableData = ({ pulls }: { pulls?: Array<Pull> }) => {
   if (!isArray(pulls)) {
     return []
   }
+
   return pulls.filter(Boolean).map((pull: Pull) => {
-    let patch, change
-    if (pull?.compareWithBase?.__typename === 'Comparison') {
-      patch = pull?.compareWithBase?.patchTotals?.percentCovered ?? 0
-      change = pull?.compareWithBase?.changeCoverage ?? 0
+    let patch = <p>-</p>
+    if (pull?.head?.coverageStatus === 'ERROR') {
+      patch = <ErroredUpload />
+    } else if (pull?.head?.coverageStatus === 'PENDING') {
+      patch = <PendingUpload />
+    } else if (
+      pull?.head?.coverageStatus === 'COMPLETED' &&
+      pull?.compareWithBase?.__typename === 'Comparison'
+    ) {
+      const percent = pull?.compareWithBase?.patchTotals?.percentCovered ?? NaN
+      patch = (
+        <TotalsNumber
+          plain={true}
+          large={false}
+          light={false}
+          value={percent}
+          showChange={false}
+        />
+      )
     }
 
     const updatestamp = pull?.updatestamp ?? undefined
     const title = pull?.title ?? 'Pull Request'
     const pullId = pull?.pullId ?? NaN
 
-    let bundleAnalysis = undefined
-    if (
-      pull?.head?.bundleAnalysisReport?.__typename === 'BundleAnalysisReport'
+    let bundleAnalysis = <p>-</p>
+    if (pull?.head?.bundleStatus === 'ERROR') {
+      bundleAnalysis = <ErroredUpload />
+    } else if (pull?.head?.bundleStatus === 'PENDING') {
+      bundleAnalysis = <PendingUpload />
+    } else if (
+      pull?.head?.bundleStatus === 'COMPLETED' &&
+      pull?.bundleAnalysisCompareWithBase?.__typename ===
+        'BundleAnalysisComparison'
     ) {
-      bundleAnalysis = <>Upload: &#x2705;</>
-    } else {
-      bundleAnalysis = <>Upload: &#x274C;</>
+      const change =
+        pull?.bundleAnalysisCompareWithBase?.bundleChange?.size.uncompress
+      const content = `${change > 0 ? '+' : ''}${formatSizeToString(change)}`
```

Comment:
if the change < 0, should we show a "-" sign?

Reply:
by default as a negative number the `-` is already present

Reply:
ah okay makes sense, I wasn't sure since the "formatSizeToString" was taking the abs of the number

Looking at your screenshot that seems to be the case, just a little hard to parse atm

Reply:
Ahh, no worries. `formatSizeToString` only abs the size to for the size checks to see if we can concatenate to a larger size (kB, MB, etc.) we than use the passed value directly in the format, rather than the absolute'ed value.